src/fireedge/src/modules/utils/helpers.js

```diff
@@ -0,0 +1,758 @@
+    .filter(
+      ({ notOnHypervisors, onlyOnHypervisors } = {}) =>
+        (!notOnHypervisors && !onlyOnHypervisors) ||
+        (notOnHypervisors && !notOnHypervisors.includes?.(hypervisor)) ||
+        onlyOnHypervisors?.includes?.(hypervisor)
+    )
+
+/**
+ * @param {Field[]} fields - Fields
+ * @param {VN_DRIVERS} driver - Driver
+ * @returns {Field[]} Filtered fields
+ */
+export const filterFieldsByDriver = (fields, driver = false) =>
+  fields
+    .map((field) => (typeof field === 'function' ? field(driver) : field))
+    .filter(
+      ({ notOnDrivers } = {}) => !driver || !notOnDrivers?.includes?.(driver)
+    )
+
+/**
+ * Filter an object list by property.
+ *
+ * @param {Array} arr - Object list
+ * @param {string} predicate - Property
+ * @returns {Array} List filtered by predicate
+ */
+export const filterBy = (arr, predicate) => {
+  const callback =
+    typeof predicate === 'function' ? predicate : (output) => output[predicate]
+
+  return [
+    ...arr
+      .reduce((map, item) => {
+        const key = item === null || item === undefined ? item : callback(item)
+
+        map.has(key) || map.set(key, item)
+
+        return map
+      }, new Map())
+      .values(),
+  ]
+}
+
+/**
+ * Get the value from an object by the path of property.
+ *
+ * @param {object} obj - Object
+ * @param {string} path - Path of property
+ * @param {*} [defaultValue] - Default value of property
+ * @returns {*} Value of property
+ */
+export const get = (obj, path, defaultValue = undefined) => {
+  const travel = (regexp) =>
+    String.prototype.split
+      .call(path, regexp)
+      .filter(Boolean)
+      .reduce(
+        (res, key) => (res !== null && res !== undefined ? res[key] : res),
+        obj
+      )
+
+  const result = travel(/[,[\]]+?/) || travel(/[,[\].]+?/)
+
+  return result === undefined || result === obj ? defaultValue : result
+}
+
+/**
+ * Deletes a given key from an object.
+ *
+ * @param {string[]} attr - Array with the path to be deleted
+ * @param {object} originalTemplate - Template to be modified
+ */
+export const deleteObjectKeys = (attr, originalTemplate = {}) => {
+  const keyToDelete = attr.pop()
+  let template = originalTemplate || {}
+  // TODO: Consider the case when restricted attributes is inside an array and goes deeper
+  attr.forEach((key) => {
+    if (Array.isArray(template?.[key])) {
+      template?.[key].forEach((element) => {
+        element && delete element[keyToDelete]
+      })
+    } else {
+      template = template[key]
+    }
+  })
+  template && delete template[keyToDelete]
+}
+
+/**
+ * Set a value of property in object by his path.
+ *
+ * @param {object} obj - Object
+ * @param {string|string[]} path - Path of property
+ * @param {*} value - New value of property
+ * @returns {object} Object with the new property
+ */
+export const set = (obj, path, value) => {
+  if (Object(obj) !== obj) return obj // When obj is not an object
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It looks like you have used `new Object()` for object creation, instead we can use `{}`. Object literals enable us to write code that supports lots of features yet still provide a relatively straightforward process and faster evaluation for the implementers of our code.

[Learn more](https://eslint.org/docs/latest/rules/no-new-object)